recipes/libpng/all/conanfile.py

```diff
@@ -51,31 +66,28 @@ def _patch(self):
                                       'COMMAND "${CMAKE_COMMAND}" -E copy_if_different $<TARGET_LINKER_FILE_NAME:${S_TARGET}> $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/${DEST_FILE}',
                                       'COMMAND "${CMAKE_COMMAND}" -E copy_if_different $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/$<TARGET_LINKER_FILE_NAME:${S_TARGET}> $<TARGET_LINKER_FILE_DIR:${S_TARGET}>/${DEST_FILE}')
 
-        if self.settings.build_type == 'Debug':
-            tools.replace_in_file(os.path.join(self._source_subfolder, 'libpng.pc.in'),
-                                  '-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@',
-                                  '-lpng@PNGLIB_MAJOR@@PNGLIB_MINOR@d')
-
     def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["PNG_TESTS"] = "OFF"
-        cmake.definitions["PNG_SHARED"] = self.options.shared
-        cmake.definitions["PNG_STATIC"] = not self.options.shared
-        cmake.definitions["PNG_DEBUG"] = "OFF" if self.settings.build_type == "Release" else "ON"
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["PNG_TESTS"] = "OFF"
```

Comment:
Maybe take this opportunity and change these to booleans? Or does the cmake script depend on `ON` and `OFF`?

Reply:
Upstream CMakeLists wants string, it fails with boolean.

Reply:
You're right. I'm looking at the cmake script now.

Reply:
fixed